src/cpp/src/utils.cpp

```diff
@@ -265,6 +267,45 @@ ov::Core singleton_core() {
     return core;
 }
 
+size_t get_first_history_difference(const ov::Tensor& encoded_history, const std::vector<int64_t> tokenized_history) {
```

Comment:
it's not vert clear how to treat the following case:
- `encoded_history` is [1, 2, 3]
- `tokenized_history` is [1, 2]
`get_first_history_difference` will return 2

and in the following case:
- `encoded_history` is [1, 2]
- `tokenized_history` is [1, 2]
`get_first_history_difference` will also return 2

But in the second case histories are the same. Maybe in case of same histories we need to return some special value like MAX size_t ? similar to `std::string::npos`

Reply:
added SIZE_MAX and merged get_first_history_difference/is_tokenized_history_same to one